### DISCUSSION TITLE
Can docker/docker-compose use the Podman Desktop registry authentication?

### DISCUSSION THREAD

**Discussion by derkoe** [podman-desktop/podman-desktop#4438] [Q&A]
When I add a custom registry under "Settings" - "Registries" I can pull images using "podman" from Windows.

But when I use "docker" or "docker-compose" from my Windows command line I get the error: 
```
unable to retrieve auth token: invalid username/password: unknown: Authentication is required
```

I can obviously use "docker login" - then the credentials are stored in my home directory in Windows and everything works. But then I have to enter the credentials twice.

Is there a way that docker/docker-compose can also use the registry auth stored in Podman Desktop?

**Comment by lclc** [comment]
I ran into the same issue on OSX. Creating a symlink fixed it for me:

`ln -s /Users/$USER/.config/containers/auth.json /Users/$USER/.docker/config.json`

**Reply by usmanzaheer1995** [reply]
I thought I was going insane, but this fixed it for me. Thank you!

**Comment by hamedrabah** [comment]
There is a supported way to avoid maintaining two independent logins, but I would not symlink the entire files: Docker and Podman can add different credential-helper fields to their configs.

The simplest shared setup on Windows is:

1. Run `docker login registry.example.com` once.
2. Let Docker store the credential in `%USERPROFILE%\.docker\config.json`.
3. Use Podman normally. Podman documents that, when the registry credential is not in its own auth file, it also checks `$HOME/.docker/config.json`.

If you prefer to perform the login with Podman, write it directly in Docker-compatible format:

```powershell
podman login --compat-auth-file "$HOME\.docker\config.json" registry.example.com
```

After that, `docker`, `docker compose`, and Podman can all consult the Docker config instead of asking you to enter the same credential twice.

The important distinction is that Podman Desktop normally writes `$HOME/.config/containers/auth.json` on Windows/macOS, while Docker reads `%USERPROFILE%\.docker\config.json`. There is no automatic UI-level synchronization between those two locations. Choosing the Docker config as the shared file is safer than replacing it with a symlink, especially when Docker is configured to use the Windows credential store.

Official references:
- Podman login auth-file lookup and `--compat-auth-file`: https://docs.podman.io/en/latest/markdown/podman-login.1.html
- Docker credential storage: https://docs.docker.com/reference/cli/docker/login/
